tictactoe/script.js

```diff
@@ -20,69 +18,54 @@ const WinCombinations = [
     [6, 4, 2]
 ];
 
-//ZMIANA NA FUNCKJE STRZALKOWA
-function checkForWin(){
+const checkForWin = () =>{
 
-    // for(let i = 0; i <=7; i++){
-    //     const [number1, number2, number3] = WinCombinations[i];
-    //     if(fields[number1] !== "" && fields[number1] === fields[number2] && fields[number1] === fields[number3]){
-    //         alert(`Player ${activePlayer} won!`);
-    //         return true;
-    //     }
-    // }
-    WinCombinations.forEach((combination)=>{
+    WinCombinations.forEach((combination)=>{ 
         const [number1, number2, number3] = combination;
-        //SPRAWDZENIE CZY POLE JEST PUSTE
-     if(fields[number1] === fields[number2] && fields[number1] === fields[number3]){
-        won = true; 
-            alert(`Player ${activePlayer} won!`);
-
-            //CO DAJE RETURN (DEBBUGER)
-            return true;
-
-        }
+        if(fields[number1] !== "" && fields[number1] === fields[number2] && fields[number1] === fields[number3]){

```

Comment:
if statement is not blocking script to run for each item even if it finds won result, add **!won &&**

Reply:
optimisation

const [firstFieldNumber] = combination || [];
const firstFieldValue = fields[firstFieldNumber];
const allFieldsAreTheSame = firstFieldValue && combination.every((fieldNumber) => fields[fieldNumber] === firstFieldValue);

if(allFieldsAreTheSame) {}